### PR TITLE
chore: update redis-ha manifests

### PR DESCRIPTION
**Pull request by crenshaw-dev**
Looks like the upstream manifests haven't been pulled in a while. Probably good to include those updates in 2.5.

### CODE REVIEW

**Comment by crenshaw-dev** [comment]
Major version announcements:
* [2.1.0](https://www.mail-archive.com/haproxy@formilux.org/msg35491.html)
* [2.2.0](https://www.mail-archive.com/haproxy@formilux.org/msg37852.html)
* [2.3.0](https://www.mail-archive.com/haproxy@formilux.org/msg38812.html)
* [2.4.0](https://www.mail-archive.com/haproxy@formilux.org/msg40499.html)
* [2.5.0](https://www.mail-archive.com/haproxy@formilux.org/msg41508.html)

Gonna read through those, then I'll bump the version.

**Comment by crenshaw-dev** [comment]
@34fathombelow could you take another look?

File: manifests/ha/base/redis-ha/chart/values.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ redis-ha:
   haproxy:
     enabled: true
     image:
-      tag: 2.0.29-alpine
+      tag: 2.5.7-alpine
```
Comment (34fathombelow) [inline manifests/ha/base/redis-ha/chart/values.yaml:12]: Upon further investigation of [HAProxy's release cycle](https://www.haproxy.org), stable version 2.5 is EOL 2023-Q1.  We should choose either 2.6(EOL 2027-Q2) or 2.4(EOL 2026-Q2) both are LTS.

Should we pin a version like we do for image builds and set the imagePullPolicy to Always? A 2.x version will always point to the the latest patch version for HAProxy.

i.e.
```yaml
 haproxy:
    enabled: true
    image:
      tag: 2.6-alpine
      pullPolicy: Always
```

**Review by 34fathombelow — CHANGES_REQUESTED** [review]

**Review by 34fathombelow — COMMENTED** [review]
Might also suggest to rename the title to something like   ```chore: update redis-ha manifests```

File: manifests/ha/base/redis-ha/chart/values.yaml (hunk 1 above)
Comment (crenshaw-dev) [reply manifests/ha/base/redis-ha/chart/values.yaml:12]: > Should we pin a version like we do for image builds and set the imagePullPolicy to Always? A 2.x version will always point to the the latest patch version for HAProxy.

Personally I like setting the whole version, making the runtime behavior of an Argo CD install more deterministic. But I could see a case either way.

> We should choose either 2.6(EOL 2027-Q2) or 2.4(EOL 2026-Q2) both are LTS.

I wonder if we should just use the most recent version of the chart which pointed at the 2.4.x series (to minimize chart/image incompatibility risk). If you're not opposed it, that's the route I'll go.

Comment (crenshaw-dev) [reply manifests/ha/base/redis-ha/chart/values.yaml:12]: @34fathombelow I'll wait for your PR to be merged and then update this one.

**Review by jannfis — APPROVED** [review]
LGTM

**Pull request merged**